code/src/App.js

```diff
@@ -1,9 +1,164 @@
-import React from 'react'
+import React, { useState } from "react";
 
 export const App = () => {
+  const [name, setName] = useState("");
+  const [superPower, setSuperPower] = useState("");
+  const [sideKick, setsideKick] = useState();
+  const [range, setRange] = useState(50);
+  const [showResult, setShowResult] = useState(false);
+  const [section, setSection] = useState("firstQuestion");
+
+  const sideKickQ = ["Oh yes I want one!", "No, I work alone", "Maybe later"];
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    setShowResult(true);
+  };
+
+  // SUMMERY------------>
   return (
-    <div>
-      Find me in src/app.js!
+    <div className="wrapper">
+      {showResult && (
+        <div className="summary">
+          <h1>Hey {name}!</h1>
+          <h4>Are you ready to save the world?</h4>
+          <p>
+            You go by - {name}
+            <br></br>
+            Your awesome superpower is - {superPower}
+            <br></br>
+            About sidekicks - {sideKick}
+            <br></br>
+            <br></br>
+            You are {range} % villain.
+            <br></br>
+            {range > 49 && "Oh, no maybe this is a bad idea!"}
+            {range < 49 && "That will do just fine."}
+          </p>
+          <button
+            id="restart"
+            type="button"
+            onClick={() => {
+              setSection("firstQuestion");
+              setShowResult(false);
+            }}
+          >
+            Restart
+          </button>
+        </div>
+      )}
+
+      {/* FIRST QUESTION -----> */}
+      {!showResult && (
+        <React.Fragment>
+          <form onSubmit={handleSubmit}>
+            <h1>Hello {name || "Stranger"}!</h1>
+            {section === "firstQuestion" && (
+              <div>
+                <label htmlFor="name">
+                  <h4>What's you superhero name?</h4>
+                </label>
+                <input
+                  id="name"
+                  required
+                  type="text"
+                  onChange={event => setName(event.target.value)}
+                  value={name}
+                />
+                <button
+                  id="first-button"
+                  type="button"
+                  onClick={() => setSection("secondQuestion")}
+                >
```

Comment:
Regarding validation: 
One thing that you can do here is to set the button to disabled if the name variable is empty. 
```
<button
    disabled={name ? false : true}   <--- If there is a name, disabled=false, else its true.
    id="first-button"
    type="button"
    onClick={() => setSection("secondQuestion")}
                >
```